draft-ietf-quic-tls.md

```diff
@@ -942,16 +942,28 @@ Packet number protection is applied after packet protection is applied (see
 encryption algorithm.
 
 In sampling the packet ciphertext, the packet number length is assumed to be the
-smaller of the maximum possible packet number encoding (4 octets), or the size
-of the protected packet minus the minimum expansion for the AEAD.  For example,
-the sampled ciphertext for a packet with a short header can be determined by:
+smaller of the maximum possible packet number encoding (4 octets), or the
+remaining space in the packet when the minimum expansion for the AEAD is
+subtracted.  For example, the sampled ciphertext for a packet with a short
+header can be determined by:
 
 ~~~
-sample_offset = min(1 + connection_id_length + 4,
+sample_offset = min(1 + len(connection_id) + 4,
                     packet_length - aead_expansion)
 sample = packet[sample_offset..sample_offset+sample_length]
 ~~~
 
+A packet with a long header is sampled in the same way, noting that multiple
+QUIC packets might be included in the same UDP datagram and that each one is
+handled separately.
+
+~~~
+sample_offset = min(2 + len(destination_connection_id) +
+                      len(source_connection_id) +
+                      len(payload_length) + 4,
+                    packet_length - aead_expansion)
```

Comment:
I assume `packet_length` refers to the QUIC packet length and not the UDP payload length, right? If so, how's this supposed to be determined from a long header packet, especially if it's the first of several compound packets?

Reply:
Ahh, I see the problem.  You don't know where the payload length starts.  It seems like we need to make the payload length cover the packet number length.  More changes inbound.